modules/affile/affile-dest.c

```diff
@@ -191,8 +203,12 @@ affile_dw_reopen(AFFileDestWriter *self)
                 evt_tag_str("filename", self->filename),
                 evt_tag_errno(EVT_TAG_OSERROR, errno),
                 NULL);
+      if(!self->reopen_pending)
+        affile_dw_arm_reopen(self);
+      g_static_mutex_unlock(&self->lock);
       return self->owner->super.super.optional;
     }
+  g_static_mutex_unlock(&self->lock);
   return TRUE;
 }
 
```

Comment:
As far as I can see now `retries()` option is not valid for `file()` destination. Simply adding this option can solve the problem without mutex magic. If `retries` tries to reach the destination in intervals specified in `time-reopen()` this patch will be a one-liner. 
We should prepare `retries` to check the destination *forever*.

```
...
time-reopen(5)
...

file( "/var/log/example.log" retries(5));

```